feat(qrcode): add hover-to-copy functionality #2851

rajanarahul93 opened the pull request:
### Describe the changes you have made in this PR

Added a hover-to-copy interaction on the QRCode component. When the user hovers over the QR code, a semi-transparent overlay appears with a copy icon and text prompting the user to click. On click, the QR code content is copied to the clipboard. This enhances UX by making the copy functionality more discoverable and user-friendly.

### Link this PR to an issue [optional]

Fixes #2851 

### Type of change

- `feat`: New feature (non-breaking change which adds functionality)

### Screenshots of the changes [optional]

![image](https://github.com/user-attachments/assets/cd136f7c-7d0e-4283-a0c9-ecec07ab4c5c)


### How has this been tested?

- Manually tested on local development environment
- Verified hover overlay appears correctly
- Verified the value is copied to clipboard on click
- Checked responsiveness across different screen sizes

### Checklist

- [x] Self-review of changed code
- [x] Manual testing
- [ ] Added automated tests where applicable
- [ ] Update Docs & Guides  
- For UI-related changes  
- [x] Darkmode  
- [x] Responsive layout

rajanarahul93 commented:
Hi @reneaaron  @stackingsaunter   
I'd like to request your review on this PR when you have time. Let me know if any changes are needed. Thanks!

pavanjoshi914 commented on src/app/components/QRCode/index.tsx:
```diff
@@ -15,21 +18,56 @@ export type Props = {
+        bgColor={bgColor}
+        className={classNames(
+          "w-full h-auto rounded-md transition-opacity",
+          isHovering ? "opacity-80" : "opacity-100",
```
for hover why we need to use state varbiles? can use tailwind here only? 

eg. 
```className="text-gray-600 hover: text-gray-800"```

pavanjoshi914 commented:
screenshot looks a bit messsy. missing translations broken ui. can you check if other elements are not effected by your ui change and update screenshots? thanks!

rajanarahul93 commented:
Thanks for your review, @pavanjoshi914 .I’ll make the suggested changes and update the PR shortly.

rajanarahul93 commented:
## UI Enhancement: Simplified QR Code Copy Interaction

### BEFORE:
- Had overlapping copy indicators:
  1. "Copy to clipboard" text + icon on hover  
  2. "Copied to clipboard" toast on click
- Dark overlay on hover made QR code less visible  
- Multiple text elements created visual noise
![image](https://github.com/user-attachments/assets/45472410-a565-47e1-b61e-63f26ceb0b99)



### AFTER:
- Clean, minimalist hover state:
  - Single copy icon with drop shadow for visibility
  - No text overlay, reducing visual clutter
  - QR code remains clearly visible with subtle opacity change
- Single feedback mechanism:
  - Toast notification on successful copy
  - Maintains accessibility without redundant UI elements
  
  
![Screenshot 2025-04-21 144409](https://github.com/user-attachments/assets/83896d98-1b9a-4676-9209-8b5036f13bd6)


###  Technical Changes:
- Removed background overlay (`bg-black bg-opacity-50`)
- Removed redundant copy text
- Added `drop-shadow-lg` to icon for better contrast
- Kept smooth opacity transitions for polish
- Maintained clipboard functionality and toast notification

@pavanjoshi914

pavanjoshi914 commented on src/app/components/QRCode/index.tsx:
```diff
@@ -21,15 +24,29 @@ export default function QRCode({ value, size, level, className }: Props) {
+        level={level}
+      />
+      <div className="absolute inset-0 flex items-center justify-center opacity-0 group-hover:opacity-100 transition-opacity">
+        <PopiconsCopyLine className="h-8 w-8 text-white drop-shadow-lg" />
```
where this is used now? i don't see any apperance of this

pavanjoshi914 commented on src/app/screens/ReceiveInvoice/index.tsx:
```diff
@@ -147,7 +147,7 @@ function ReceiveInvoice() {
                   <QRCode
                     value={invoice.paymentRequest.toUpperCase()}
-                    size={512}
+                    size={192}
```
revert

pavanjoshi914 commented:
![image](https://github.com/user-attachments/assets/548dad8b-afab-4e90-9173-543f2743de7b)
hovering over this part, qr code content aint' copying. 

@stackingsaunter can you please assit him in hover opacity and overlay that should be shown when user hovers the qr code. currently its not that clear visually that we can copy qr code

rajanarahul93 commented:
@pavanjoshi914 , I will make sure the required changes are done!!

rajanarahul93 commented:
@pavanjoshi914 any suggestions?

rajanarahul93 commented:
@stackingsaunter any suggestions from you side to solve the issue?
